.changeset/curvy-otters-jog.md

```diff
@@ -0,0 +1,18 @@
+---
+'astro': minor
+'@astrojs/db': minor
```

Comment:
We should create a new changeset only for `@astrojs/db` that shows the migration path of the new type. Since this is a breaking change, I believe this should stand out in the changelog

Reply:
Just noting that I'll review here when I have a docs PR!

Reply:
It is barely even a breaking change for `@astrojs/db` though. It breaks something we never said was there (the `AstroDbIntegration` type). The helper `defineDbIntegration` method [shown on the docs](https://docs.astro.build/en/guides/astro-db/#building-astro-db-integrations) continues to work with no changes.

The main difference is requiring the new version of `astro`.

But I'll write that up either way